Terminology cleanup: Controller / Built-In Node

daniel-beck opened the pull request:
Actually implements the decision from mid last year now that the necessary followup has finally happened.

<details>
<summary>Screenshots</summary>

### Executors widget

> <img width="569" alt="Executors widget" src="https://user-images.githubusercontent.com/1831569/117355778-467e6680-aeb3-11eb-9fb7-073b41e8990d.png">

### Built-in node UI

> <img width="1838" alt="Built-in node UI" src="https://user-images.githubusercontent.com/1831569/117355787-48482a00-aeb3-11eb-8523-c62ed2967dfc.png">

### Way too wordy

> <img width="1635" alt="Way too wordy" src="https://user-images.githubusercontent.com/1831569/117355795-4a11ed80-aeb3-11eb-8b85-d3da0bcf8523.png">

</details>

TODO list:

- [x] Put final version number in `buildEnv.properties`

### Proposed changelog entries

* Replace the term "master" with "controller" (for the main Jenkins application) or "built-in node", as appropriate.
* Add migration code to only change the node name (e.g. `NODE_NAME` environment variable) and label of the built-in node after explicit migration by an administrator. New installations get the new node and and label immediately.
* Add the system property `jenkins.model.Jenkins.nodeNameAndSelfLabelOverride` to specify a different node name and label for the built-in node (e.g. for Configuration as Code use cases) than the one otherwise determined. This will not affect other uses of the node name, such as the URL to the built-in node (now `/computer/(built-in)/`).

<!-- Comment: 
The changelogs will be integrated by the core maintainers after the merge.  See the changelog examples here: https://jenkins.io/changelog/ -->

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

<!-- For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc) -->

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/code-reviewers
-->

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

daniel-beck committed:
Terminology cleanup: Controller (mostly)

daniel-beck committed:
Fix NPE

daniel-beck committed:
Fix some tests

daniel-beck committed:
Add tool installers to impact, slight rephrasing

daniel-beck committed:
More cleanup

timja commented on core/src/main/java/hudson/model/Computer.java:
```diff
@@ -480,7 +480,7 @@ public final long getConnectTime() {
      * Disconnect this computer.
      *
-     * If this is the master, no-op. This method may return immediately
+     * If this is the blub node, no-op. This method may return immediately
```
blub 😂

daniel-beck replied:
8 months without followup 🤷

daniel-beck committed:
Rename help files for renamed field

oleg-nenashev commented:
FTR new thread for Jenkins controller sub-terms (e.g. Jenkins as a main node, etc.) is here: https://groups.google.com/u/1/g/jenkinsci-dev/c/x5vdlJDvntw . I am taking over this 8-months-old action item from other contributors so we can finally get it over the line. Any contributions and feedback are welcome!

daniel-beck committed:
Make TcpSlaveAgentListener available at tcp-agent-listener/ URL

daniel-beck committed:
Merge branch 'master' into controller

daniel-beck committed:
Noting future changes to do, replace deprecated getter access

daniel-beck committed:
Some more cleanup

oleg-nenashev commented:
FTR https://github.com/jenkins-infra/jenkins.io/issues/4332 for the glossary update

daniel-beck committed:
Blub -> Built-in

daniel-beck committed:
Add recommendation for label assignments

daniel-beck committed:
Merge branch 'master' into controller

daniel-beck committed:
Merge remote-tracking branch 'jenkinsci/master' into controller

daniel-beck committed:
More cleanup

daniel-beck committed:
Rename resource files

daniel-beck committed:
Update labels when migrating

daniel-beck committed:
Jenkins singleton is created differently from other objects

This trick doesn't seem to work, so do a version comparison :-(

daniel-beck committed:
This sucks

daniel-beck committed:
Hudson being special is annoying

daniel-beck committed:
Newly set up instances do not need migration

daniel-beck committed:
Allow setting a system property to override the node label

This should help legacy build scripts running on CasC based instances
while not having an active admin monitor

daniel-beck committed:
Fix test

StackScribe commented on core/src/main/java/hudson/model/AbstractCIBase.java:
```diff
@@ -147,7 +147,7 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
             }
         } else {
-            // we always need Computer for the master as a fallback in case there's no other Computer.
+            // we always need Computer for the built-in node as a fallback in case there's no other Computer.
```
```suggestion
            // we always need a Computer for the built-in node as a fallback,  in case there's no other Computer.
```

StackScribe commented on core/src/main/java/hudson/model/Computer.java:
```diff
@@ -142,7 +142,7 @@
  * This object is related to {@link Node} but they have some significant differences.
```
```suggestion
 * This object is related to {@link Node} but has some significant differences.
```

StackScribe commented on core/src/main/java/hudson/model/Computer.java:
```diff
@@ -142,7 +142,7 @@
  * This object is related to {@link Node} but they have some significant differences.
  * {@link Computer} primarily works as a holder of {@link Executor}s, so
```
```suggestion
 * {@link Computer} primarily works as a holder of {@link Executor}s, so,
```

StackScribe commented on core/src/main/java/hudson/model/Computer.java:
```diff
@@ -142,7 +142,7 @@
  * {@link Computer} primarily works as a holder of {@link Executor}s, so
  * if a {@link Node} is configured (probably temporarily) with 0 executors,
- * you won't have a {@link Computer} object for it (except for the master node,
+ * you won't have a {@link Computer} object for it (except for the built-in node,
```
```suggestion
 * you don't have a {@link Computer} object for it (other the built-in node,
```

StackScribe commented on core/src/main/java/hudson/model/Computer.java:
```diff
@@ -1656,7 +1656,7 @@ public boolean isAcceptingTasks() {
     @CLIResolver
     public static Computer resolveForCLI(
-            @Argument(required=true,metaVar="NAME",usage="Agent name, or empty string for master") String name) throws CmdLineException {
+            @Argument(required=true,metaVar="NAME",usage="Agent name, or empty string for built-in node") String name) throws CmdLineException {
```
Is something missing after "throws"?

daniel-beck replied:
What would be missing?

StackScribe commented on core/src/main/java/hudson/model/DirectoryBrowserSupport.java:
```diff
@@ -525,7 +525,7 @@ private static void sendZipUsingListOfNames(ZipOutputStream zos, VirtualFile dir
         // In ZIP archives "All slashes MUST be forward slashes" (http://pkware.com/documents/casestudies/APPNOTE.TXT)
         // TODO On Linux file names can contain backslashes which should not treated as file separators.
-        //      Unfortunately, only the file separator char of the master is known (File.separatorChar)
+        //      Unfortunately, only the file separator char of the controller is known (File.separatorChar)
         //      but not the file separator char of the (maybe remote) "dir".
```
```suggestion
        //      Unfortunately, only the file separator char of the controller is known (File.separatorChar);
        //      the file separator char of the (maybe remote) "dir" is not known.
```

StackScribe commented on core/src/main/java/hudson/model/Slave.java:
```diff
@@ -115,7 +115,7 @@
     /**
      * Path to the root of the workspace from the view point of this node, such as "/hudson", this need not
-     * be absolute provided that the launcher establishes a consistent working directory, such as "./.jenkins-slave"
+     * be absolute provided that the launcher establishes a consistent working directory, such as "./.jenkins-agent"
```
```suggestion
     * be absolute, provided that the launcher establishes a consistent working directory, such as "./.jenkins-agent"
```

StackScribe commented on core/src/main/java/hudson/model/WorkspaceBrowser.java:
```diff
@@ -34,7 +34,7 @@
  * Allows to access a workspace as an alternative to online build node.
```
```suggestion
 * Allows access to a workspace as an alternative to online build node.
```

StackScribe commented on core/src/main/java/hudson/slaves/SlaveComputer.java:
```diff
@@ -1073,12 +1073,12 @@ public Void call() {
-     * Obtains a {@link VirtualChannel} that allows some computation to be performed on the master.
-     * This method can be called from any thread on the master, or from agent (more precisely,
+     * Obtains a {@link VirtualChannel} that allows some computation to be performed on the controller.
+     * This method can be called from any thread on the controller, or from agent (more precisely,
```
```suggestion
     * This method can be called from any thread on the controller, or from an agent (more precisely,
```

StackScribe commented on core/src/main/java/hudson/slaves/SlaveComputer.java:
```diff
@@ -1073,12 +1073,12 @@ public Void call() {
-     * This method can be called from any thread on the master, or from agent (more precisely,
+     * Obtains a {@link VirtualChannel} that allows some computation to be performed on the controller.
+     * This method can be called from any thread on the controller, or from agent (more precisely,
      * it only works from the remoting request-handling thread in agents, which means if you've started
      * separate thread on agents, that'll fail.)
```
```suggestion
     * it only works from the remoting request-handling thread in agents, which means that,
     * if you start a separate thread on an agent, it fails.)
```

StackScribe commented on core/src/main/java/hudson/tools/ToolDescriptor.java:
```diff
@@ -173,7 +173,7 @@ public FormValidation doCheckHome(@QueryParameter File value) {
     /**
      * May be overridden to provide tool-specific validation of a tool home directory.
-     * @param home a possible value for {@link ToolInstallation#getHome}, known to already exist on the master
+     * @param home a possible value for {@link ToolInstallation#getHome}, known to already exist on the controller TODO or is it built-in?
```
```suggestion
     * @param home a possible value for {@link ToolInstallation#getHome}, known to already exist on the controller //TODO or is it built-in?
```

daniel-beck replied:
Eh, we're still in a comment :)

StackScribe commented on core/src/main/java/hudson/tools/ToolInstallation.java:
```diff
@@ -141,7 +141,7 @@ public String getName() {
      * 
      * The path can be in Unix format as well as in Windows format.
```
```suggestion
     * The path can be in either Unix format or Windows format.
```

StackScribe commented on core/src/main/java/jenkins/model/Jenkins.java:
```diff
@@ -382,7 +382,7 @@
 
     /**
-     * Number of executors of the master node.
+     * Number of executors of the built-in node.
```
```suggestion
     * Number of executors on the built-in node.
```
or maybe "for the built-in node"?

StackScribe replied:
```suggestion
     * Number of executors on the built-in node.
```

StackScribe commented on core/src/main/java/jenkins/model/Jenkins.java:
```diff
@@ -824,11 +827,11 @@ public static Jenkins getActiveInstance() throws IllegalStateException {
      * For those rare cases use this method.
-     * <p>In other cases you may have code that might end up running on a remote JVM and not on the Jenkins master.
+     * <p>In other cases you may have code that might end up running on a remote JVM and not on the Jenkins controller or built-in node.
      * For those cases you really should rewrite your code so that when the {@link Callable} is sent over the remoting channel
```
```suggestion
     * For those cases, you really should rewrite your code so that, when the {@link Callable} is sent over the remoting channel,
```

StackScribe replied:
```suggestion
     * For those cases, you really should rewrite your code so that, when the {@link Callable} is sent over the remoting channel,
```

StackScribe commented on core/src/main/java/jenkins/model/Jenkins.java:
```diff
@@ -863,6 +866,13 @@ public static Jenkins getInstance() {
 
+    /**
+     * If this is false, no migration is needed to reconfigure the built-in node (formerly 'master', now 'built-in').
+     * Otherwise, {@link BuiltInNodeMigration} will show up.
```
```suggestion
     * Otherwise, {@link BuiltInNodeMigration} shows up.
```

StackScribe commented on core/src/main/java/jenkins/model/Jenkins.java:
```diff
@@ -1293,10 +1309,10 @@ public void setAgentProtocols(Set<String> protocols) {
-                //    have the protocol disabled for them. This is what we want. If they didn't upgrade to a master
+                //    the initial opt-in admins, assuming they visited an upgrade to a controller with step 2, will
+                //    have the protocol disabled for them. This is what we want. If they didn't upgrade to a controller
                 //    with step 2, well there is not much we can do to differentiate them from somebody who is upgrading
```
```suggestion
                //    with step 2, we are not able to differentiate them from somebody who is upgrading
```

StackScribe replied:
```suggestion
                //    with step 2, we are not able to differentiate them from somebody who is upgrading
```

StackScribe commented on core/src/main/java/jenkins/model/Jenkins.java:
```diff
@@ -1293,10 +1309,10 @@ public void setAgentProtocols(Set<String> protocols) {
-                //    from a previous step 3 master and had needed to keep the protocol turned on.
+                //    from a previous step 3 controller and had needed to keep the protocol turned on.
                 //
                 // What we should never do is flip-flop: opt-in -> opt-out -> opt-in -> opt-out as that will basically
                 // clear any preference that an admin has set, but this should be ok as we only ever will be
```
```suggestion
                // What we should never do is flip-flop: opt-in -> opt-out -> opt-in -> opt-out as that basically
                // clears any preference that an admin has set, but this should because we will only be
```

StackScribe replied:
```suggestion
                // What we should never do is flip-flop: opt-in -> opt-out -> opt-in -> opt-out as that basically
                // clears any preference that an admin has set, but this should be okay because we will only be
```

StackScribe commented on core/src/main/java/hudson/tools/ToolDescriptor.java:
```diff
@@ -173,7 +173,7 @@ public FormValidation doCheckHome(@QueryParameter File value) {
     /**
      * May be overridden to provide tool-specific validation of a tool home directory.
-     * @param home a possible value for {@link ToolInstallation#getHome}, known to already exist on the master
+     * @param home a possible value for {@link ToolInstallation#getHome}, known to already exist on the controller TODO or is it built-in?
```
```suggestion
     * @param home a possible value for {@link ToolInstallation#getHome}, known to already exist on the controller  //TODO or is it built-in?
```

StackScribe commented on core/src/main/java/jenkins/model/Jenkins.java:
```diff
@@ -824,11 +827,11 @@ public static Jenkins getActiveInstance() throws IllegalStateException {
      * {@link #get} is what you normally want.
      * <p>In certain rare cases you may have code that is intended to run before Jenkins starts or while Jenkins is being shut down.
      * For those rare cases use this method.
```
```suggestion
     * For those rare cases, use this method.
```

StackScribe commented on core/src/main/resources/hudson/model/PasswordParameterDefinition/help.html:
```diff
@@ -1,5 +1,5 @@
     Pass a password to your build.
     The password entered here is made available to the build in plain text as an environment variable like a string parameter would be.
-    The value will be stored encrypted on the Jenkins master, similar to passwords in Jenkins configuration.
+    The value will be stored encrypted on the Jenkins controller, similar to passwords in Jenkins configuration.
```
```suggestion
    The value is stored encrypted on the Jenkins controller, similar to passwords in Jenkins configuration.
```

StackScribe commented on core/src/main/resources/hudson/slaves/JNLPLauncher/help.properties:
```diff
@@ -21,14 +21,14 @@
   In one mode, <a href="https://en.wikipedia.org/wiki/Java_Web_Start" rel="noopener noreferrer" target="_blank">Java Web Start</a> is used. \
   In this case, a JNLP file must be opened on the agent machine, \
-  which will establish a TCP connection to the Jenkins master. \
+  which will establish a TCP connection to the Jenkins controller. \
```
```suggestion
  which establishes a TCP connection to the Jenkins controller. \
```

StackScribe commented on core/src/main/resources/hudson/slaves/JNLPLauncher/help.properties:
```diff
@@ -21,14 +21,14 @@
+  the agent just needs to be able to reach the controller. \
   If you have enabled security via the <i>Configure Global Security</i> page, \
-  you can customize the port on which the Jenkins master will listen for incoming agent connections.<br>\
+  you can customize the port on which the Jenkins controller will listen for incoming agent connections.<br>\
```
```suggestion
  you can customize the port on which the Jenkins controller listens for incoming agent connections.<br>\
```

StackScribe commented on core/src/main/resources/hudson/slaves/JNLPLauncher/help.properties:
```diff
@@ -21,14 +21,14 @@
   If you have enabled security via the <i>Configure Global Security</i> page, \
-  you can customize the port on which the Jenkins master will listen for incoming agent connections.<br>\
+  you can customize the port on which the Jenkins controller will listen for incoming agent connections.<br>\
   By default, the agent will launch a GUI, but it is also possible to run \
```
```suggestion
  By default, the agent launches a GUI, but it is also possible to run \
```

daniel-beck commented:
Marked grammar feedback unrelated to the changes in this PR as resolved after discussion with @StackScribe out of band, as an attempt to contain the scope to actual terminology cleanup (which is already wider than it needs to be for "master").

daniel-beck committed:
Merge branch 'master' into controller

daniel-beck committed:
Fix test failure due to mismatched Channel property

Also add 'agent' method in AgentComputerUtil and deprecate 'slave' method

daniel-beck committed:
Add migration test

daniel-beck committed:
Add test ensuring the system property works

daniel-beck committed:
Use proper at-since format

daniel-beck commented on core/src/main/java/jenkins/security/s2m/AdminCallableMonitor.java:
```diff
@@ -30,7 +30,7 @@
 
     public AdminCallableMonitor() {
-        super("slaveToMasterAccessControl");
+        super("agentToControllerAccessControl");
```
TODO: This might be incompatible.

daniel-beck replied:
It is.

daniel-beck commented on core/src/main/resources/jenkins/model/CoreEnvironmentContributor/buildEnv.properties:
```diff
@@ -10,14 +10,14 @@ EXECUTOR_NUMBER.blurb=\
   carrying out this build. This is the number you see in \
   the "build executor status", except that the number starts from 0, not 1.
-NODE_NAME.blurb=Name of the agent if the build is on an agent, or "master" if run on master.
+NODE_NAME.blurb=Name of the agent if the build is on an agent, or "built-in" if run on the built-in node (or "master" until Jenkins 2.TODO).
```
Update once we know the version.

timja replied:
```suggestion
NODE_NAME.blurb=Name of the agent if the build is on an agent, or "built-in" if run on the built-in node (or "master" until Jenkins 2.307).
```

daniel-beck commented on core/src/main/resources/lib/hudson/executors.properties:
```diff
@@ -1,3 +1,4 @@
-Computers=master{0,choice,0#|1# + {0,number} computer ({1} of {2} executors)|1< + {0,number} computers ({1} of {2} executors)}
+# TODO this label is misleading when there are no executors configured for the built-in node
```
TODO: Review how to change this.

daniel-beck committed:
Remove tcp-agent-listener again, this is better delivered independently

daniel-beck committed:
Explicitly mark TODO comments for this change

daniel-beck commented:
This should be ready for review now. A few specific things I plan to still do in the scope of this PR. Not all TODOs I noted here need to be done in this PR (e.g. `slaves/` log directory name migration).

daniel-beck committed:
Make this confusing naming clear in Javadoc

Wadeck commented on core/src/main/resources/jenkins/model/Messages.properties:
```diff
@@ -41,7 +41,7 @@ Hudson.NotUsesUTF8ToDecodeURL=\
   this will cause problems. \
   See <a href="https://jenkins.io/redirect/troubleshooting/utf8-url-decoding">Tomcat i18n</a> for more details.
-Hudson.NodeDescription=the master Jenkins node
+Hudson.NodeDescription=the Jenkins controller''s built-in node
```
(for other reviewers)

![Screenshot_2021-05-18_101012_001](https://user-images.githubusercontent.com/2662497/118615731-4f2f4080-b7c1-11eb-8de6-f5d6500bfb70.png)

it's working fine ^^

Wadeck commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -1,6 +1,6 @@
 # The MIT License
```
FTR the other node_xx.properties were deleted, and one was just renamed. Perhaps a bit weird in isolation but easy to understand with the big picture

Wadeck reviewed (approved):
Manually tested some parts, not tested the migration.

Good job with the compatible approach 👍

aHenryJard commented on core/src/main/resources/hudson/tools/Messages.properties:
```diff
@@ -30,5 +30,5 @@ ZipExtractionInstaller.bad_connection=Server rejected connection.
 ZipExtractionInstaller.could_not_connect=Could not connect to URL.
 InstallSourceProperty.DescriptorImpl.displayName=Install automatically
-ToolDescriptor.NotADirectory={0} is not a directory on the Jenkins master (but perhaps it exists on some agents)
+ToolDescriptor.NotADirectory={0} is not a directory on the Jenkins controller (but perhaps it exists on some agents)
```
same for "Jenkins controller"

aHenryJard commented on core/src/main/resources/hudson/tools/ZipExtractionInstaller/help-url.html:
```diff
@@ -3,5 +3,5 @@
     The timestamp on the server will be compared to the local version (if any)
     so you can publish updates easily.
-    The URL must be accessible from the Jenkins master but need not be accessible from agents.
+    The URL must be accessible from the Jenkins controller but need not be accessible from agents.
```
same for "Jenkins controller"

aHenryJard reviewed (approved):
Huge work, thank you, and with great test coverage. I only have minor comment/not blocker comments so I approve it now.

daniel-beck commented on core/src/main/java/jenkins/model/Jenkins.java:
```diff
@@ -2177,7 +2194,7 @@ public Cloud getCloud(String name) {
     /**
      * Returns all {@link Node}s in the system, excluding {@link Jenkins} instance itself which
-     * represents the master.
+     * represents the built-in node (in other words, this only returns agents).
```
The built-in node is a node, but not an agent. It's explicitly excluded here.

daniel-beck commented on core/src/main/resources/hudson/slaves/JNLPLauncher/help.properties:
```diff
@@ -21,14 +21,14 @@
 
 blurb=\
-  Allows an agent to be connected to the Jenkins master whenever it is ready.<br>\
+  Allows an agent to be connected to the Jenkins controller whenever it is ready.<br>\
```
> it's either a "Jenkins instance" or a "controller" (without Jenkins) usually

[citation needed]

A text quick search on jenkins-infra/jenkins.io shows that it's almost evenly split (i.e. half of "controller" occurrences have a "Jenkins" before them). Granted, some of that is influenced by security advisories where we put "Jenkins controller", but there's lots of occurrences elsewhere. For "Jenkins master"/"master" before it, there was also no clear preference -- since these changes neither add nor remove the "Jenkins" prefix.

aHenryJard replied:
ok, thank you for the deep analyze, it's not a blocker anyway, more a personal preference :). I am fine with the your current version.

daniel-beck commented on core/src/main/java/jenkins/model/Jenkins.java:
```diff
@@ -2034,7 +2050,8 @@ public boolean isUpgradedFromBefore(VersionNumber v) {
     public @CheckForNull Computer getComputer(@Argument(required=true,metaVar="NAME",usage="Node name") @NonNull String name) {
-        if(name.equals("(master)"))
+        if(name.equals("(built-in)")
+                || name.equals("(master)")) // backwards compatibility for URLs
```
Too disconnected (logs for admins, users go to some URL) to be useful, I think.

daniel-beck commented on core/src/main/java/jenkins/agents/AgentComputerUtil.java:
```diff
@@ -35,26 +35,36 @@ private AgentComputerUtil() {
         // if this method is called from within the agent computation thread, this should work
         Channel c = Channel.current();
-        if (c != null && Boolean.TRUE.equals(c.getProperty("slave"))) {
+        if (c != null && Boolean.TRUE.equals(c.getProperty("agent"))) {
```
Channels don't survive restarts and both the setter and getter calls are updated. Only a handful of plugins call
```
hudson/remoting/Channel#setProperty
hudson/remoting/Channel#getProperty
```
and none of them use this key.
I would be surprised if we needed this.

daniel-beck committed:
Merge branch 'master' into controller

daniel-beck committed:
Merge remote-tracking branch 'jenkinsci/master' into controller

daniel-beck committed:
Rename some variables

amuniz commented on core/src/main/java/jenkins/agents/AgentComputerUtil.java:
```diff
@@ -35,26 +35,36 @@ private AgentComputerUtil() {
+
+    /**
+     * @deprecated Use {{@link #getChannelToController()}}.
+     * @since 2.235
```
2.235?

daniel-beck replied:
Yes, this is the method that previously existed and is retained for API compatibility. The body and original Javadoc moved over with minor modifications.

Vlatombe reviewed (approved):
looking good so far.

mikecirioli commented on core/src/main/java/hudson/slaves/SlaveComputer.java:
```diff
@@ -816,7 +816,7 @@ public void tryReconnect() {
     }
 
-    @WebMethod(name="slave-agent.jnlp")
+    @WebMethod(name="slave-agent.jnlp") // backward compatibility
```
Maybe log something here, i assume it won't be backward compatible forever?

daniel-beck replied:
Would keep this around at least for a few years, as it's not expensive to support additional URLs. I added this comment to prevent people from search & replacing this accidentally.

mikecirioli commented on core/src/main/java/hudson/slaves/SlaveComputer.java:
```diff
@@ -832,7 +832,7 @@ public HttpResponse doJenkinsAgentJnlp(StaplerRequest req, StaplerResponse res)
         }
 
-        @WebMethod(name="slave-agent.jnlp")
+        @WebMethod(name="slave-agent.jnlp") // backward compatibility
```
Maybe log something here?  I assume it won't be backward compatible forever

daniel-beck committed:
Merge branch 'master' into controller

daniel-beck committed:
Add BuiltInNodeMigration i18n

daniel-beck committed:
Revert admin monitor ID change

daniel-beck commented on core/src/main/java/hudson/model/Computer.java:
```diff
@@ -1677,6 +1677,7 @@ public static Computer resolveForCLI(
      *
      * @see #getLogFile()
      */
+    // TODO(terminology) migrate from slaves/ to agents/
```
We already have too many tickets nobody looks at, it's unclear to me what that would accomplish.

daniel-beck committed:
Don't fall back to migration mode on restart

daniel-beck committed:
Improve collapsed executor widget label

daniel-beck commented:
Just one TODO left -- how to combine `@LocalData` and `JenkinsSessionRule`, plus inserting the file version number this change is introduced in a resource file (which can be done by the person merging).

daniel-beck committed:
Do not use LocalData to provide JENKINS_HOME content to support restarts

daniel-beck commented:
I have a way to test the expected behavior with restarts.

I consider this PR done now. The single TODO item in the original comment needs to be done once we know which weekly this goes in.

Dan-Heath commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -20,4 +20,10 @@
 # THE SOFTWARE.
 
-master=master
+blurb = The term "master" for the main process of Jenkins and built-in node is being retired. \
```
```suggestion
blurb = The word "master" is being retired as the term for the main Jenkins process and the built-in node. \
```

Dan-Heath commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -20,4 +20,10 @@
-master=master
+blurb = The term "master" for the main process of Jenkins and built-in node is being retired. \
+  The main process is now called "controller" and the built-in node is called just "built-in node". \
+  The latter change also changes the name-derived label assigned to that node (<code>master</code> to <code>built-in</code>) and the <code>NODE_NAME</code> \
```
```suggestion
  Further, the built-in node label is changing from <code>master</code> to <code>built-in</code>, and the <code>NODE_NAME</code> \
```

Dan-Heath commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -20,4 +20,10 @@
+blurb = The term "master" for the main process of Jenkins and built-in node is being retired. \
+  The main process is now called "controller" and the built-in node is called just "built-in node". \
+  The latter change also changes the name-derived label assigned to that node (<code>master</code> to <code>built-in</code>) and the <code>NODE_NAME</code> \
+  environment variable passed to builds running on that node. \
```
```suggestion
  environment variable is passed to builds running on that node. \
```

daniel-beck replied:
This is ungrammatical with the complete sentence.

Dan-Heath commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -20,4 +20,10 @@
+  The main process is now called "controller" and the built-in node is called just "built-in node". \
+  The latter change also changes the name-derived label assigned to that node (<code>master</code> to <code>built-in</code>) and the <code>NODE_NAME</code> \
+  environment variable passed to builds running on that node. \
+  Depending on your configuration, this is a potentially breaking change, so it is not applied automatically. \
```
```suggestion
  This is potentially a breaking change, so it is not applied automatically. \
```

Dan-Heath commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -20,4 +20,10 @@
+  The latter change also changes the name-derived label assigned to that node (<code>master</code> to <code>built-in</code>) and the <code>NODE_NAME</code> \
+  environment variable passed to builds running on that node. \
+  Depending on your configuration, this is a potentially breaking change, so it is not applied automatically. \
+  Review label assignments in job configurations and tool installers, and use of the <code>NODE_NAME</code> environment variable in build scripts before applying this change. \
```
```suggestion
  Before you apply this change, review any label assignments in job configurations and tool installers, and use of the <code>NODE_NAME</code> environment variable in build scripts to identify potential issues. \
```

Dan-Heath commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -20,4 +20,10 @@
+  environment variable passed to builds running on that node. \
+  Depending on your configuration, this is a potentially breaking change, so it is not applied automatically. \
+  Review label assignments in job configurations and tool installers, and use of the <code>NODE_NAME</code> environment variable in build scripts before applying this change. \
+  Most problems with label assignments can likely be worked around by explicitly adding the <code>master</code> label to the built-in node.
```
```suggestion
  You can resolve most issues with label assignments by explicitly adding the <code>master</code> label to the built-in node.
```

Dan-Heath reviewed (approved):
I made a few suggestions for the admin monitor message. Please let me know if you have any questions, or if there's anything I can do to help.

kataggart commented:
Did what I could given my level of knowledge and I didn't see anything to be changed. Thanks @Dan-Heath for your suggested changes on the disclaimer/notification. Thanks all for doing this.

mikecirioli commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -20,4 +20,10 @@
+  The latter change also changes the name-derived label assigned to that node (<code>master</code> to <code>built-in</code>) and the <code>NODE_NAME</code> \
+  environment variable passed to builds running on that node. \
+  Depending on your configuration, this is a potentially breaking change, so it is not applied automatically. \
+  Review label assignments in job configurations and tool installers, and use of the <code>NODE_NAME</code> environment variable in build scripts before applying this change. \
```
Should we advise to take a backup first as well?

daniel-beck replied:
System property override allows using `master` even after migration is done.

daniel-beck committed:
Merge remote-tracking branch 'jenkinsci/master' into controller

daniel-beck committed:
Replace deprecated Hudson#getSlaves() call in favor of Jenkins#getNodes()

daniel-beck committed:
Merge remote-tracking branch 'jenkinsci/master' into controller

daniel-beck committed:
Rewrote admin monitor to more easily digestible lists

daniel-beck commented:
Rewrote the admin monitor to be less of a wall of text, adopting some of the suggestions:

> <img width="1768" alt="Screenshot" src="https://user-images.githubusercontent.com/1831569/126939676-b8f85bf8-7126-49da-909f-2daf392df31b.png">

daniel-beck committed:
Forgot to commit the link

timja commented on core/src/main/resources/jenkins/model/BuiltInNodeMigration/message.properties:
```diff
@@ -0,0 +1,40 @@
+  </li>\
+    <li>Review use of the <code>NODE_NAME</code> environment variable in build scripts.</li>\
+  </ul> \
+  <a href="https://www.jenkins.io/redirect/built-in-node-migration" rel="noopener noreferrer" target="_blank">Learn more.</a>
```
I can't see a PR for this?

daniel-beck replied:
Still local only, had some trouble with Docker yesterday. But yes, this will have a linked PR.

daniel-beck replied:
https://github.com/jenkins-infra/jenkins.io/pull/4496

timja commented:
Given 4 approvals I think we can go ahead with this?

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

daniel-beck commented:
Reminder that, before being merged, the version number in `buildEnv.properties` needs to be filled in.

timja commented:
> Reminder that, before being merged, the version number in `buildEnv.properties` needs to be filled in.

@daniel-beck it looks like maintainers don't have access to your fork, could you either update the version or it can be done post merge

oleg-nenashev reviewed (approved):
I do not see any obvious issues with this pull request. +1 for merging. Should there be any compatibility issues, we can clean them up before the next LTS baseline.

timja commented:
> Reminder that, before being merged, the version number in `buildEnv.properties` needs to be filled in.

https://github.com/jenkinsci/jenkins/commit/8e768057d772eb1107fc3e85c5164f01d88604dd

daniel-beck commented:
@timja 

Looks like 2.307 is the first version with this change? If so, the help text should refer to 2.306.

timja commented:
Thanks, corrected in https://github.com/jenkinsci/jenkins/commit/a621460bccea4942ad328098e4c35d7adbf6dae7

basil commented:
Caused extensive breakage in PCT/BOM tests.

daniel-beck commented:
@basil Thanks for filing the PRs. Are you taking care of all of the regressions? If there's still more broken, I'd be happy to file fixes tomorrow.

basil commented:
I am not taking care of _all_ of the regressions. I started helping out by filing PRs but ran out of time before I got to the end. For example [this run](https://ci.jenkins.io/job/Tools/job/bom/job/PR-640/5/) shows some new failures in `org.jenkinsci.plugins.workflow.EnvWorkflowTest` that remain unaddressed. There may be additional failures that I haven't noticed yet, too.
